src/test/uint256_tests.cpp

```diff
@@ -166,12 +166,14 @@ BOOST_AUTO_TEST_CASE( methods ) // GetHex SetHex begin() end() size() GetLow64 G
     uint256 TmpL(R1L);
     BOOST_CHECK_EQUAL(TmpL, R1L);
     // Verify previous values don't persist when setting to truncated string.
-    TmpL.SetHex("21");
+    TmpL.SetHexDeprecated("21");
     BOOST_CHECK_EQUAL(TmpL.ToString(), "0000000000000000000000000000000000000000000000000000000000000021");
-    TmpL.SetHex(R2L.ToString());   BOOST_CHECK_EQUAL(TmpL, R2L);
-    TmpL.SetHex(ZeroL.ToString()); BOOST_CHECK_EQUAL(TmpL, uint256());
+    TmpL = *uint256::FromHex(R2L.ToString());
+    BOOST_CHECK(TmpL == R2L);
+    TmpL = *uint256::FromHex(ZeroL.ToString());
+    BOOST_CHECK(TmpL == uint256());
 
-    TmpL.SetHex(R1L.ToString());
+    TmpL = *uint256::FromHex(R1L.ToString());
```

Comment:
Not sure whether we should only do `FromHex` testing or do it in addition to `SetHexDeprecated` testing.

Sure, we want to remove `SetHexDeprecated` but it is unclear when or how easily it could be done, do you have an exploratory branch doing the removal?

Reply:
There's only 1 remaining explicit call to `SetHexDeprecated` in this test case, before there were 7 calls to `SetHex`. But I guess almost the same tests are done indirectly in the `basics` test case above where `SetHexDeprecated` is being called via `uint<N>S`.